### PR TITLE
DRILL-8109: Hive storage plugin support reading parquet timestamp type with int64 logical type

### PR DESCRIPTION
**Pull request by cdmikechen**
# [DRILL-8109](https://issues.apache.org/jira/browse/DRILL-8109): Drill can not read parquet timestamp type with logical type in hive storage plugin

## Description

Parquet timestamp int96 had been deprecated.
https://github.com/apache/parquet-format/pull/86/files

Many computing engines that support parquet reading (such as Hive4, Hudi, Spark and so on) can use Int64 as the time storage type. When using these computing engines to read parquet files, we should also support long data type reading.

We should support int64 with logical type timestamp by hive storage.

## Documentation
(Please describe user-visible changes similar to what should appear in the Drill documentation.)

## Testing
(Please describe how this PR has been tested.)

### CODE REVIEW

**Review by cgivre — CHANGES_REQUESTED** [review]
@cdmikechen Thanks for this contribution.   Could you please add a unit test for this?

**Comment by vvysotskyi** [comment]
@cdmikechen, please make sure that this change doesn't break the support for Hive 2. If I recall correctly, `LongWritable` was added in Hive 3. Please make sure that after your changes Drill is able to build with the following extra properties:
`-Dhive.version=2.3.2 -Dfreemarker.conf.file=src/main/codegen/config.fmpp`

**Comment by cdmikechen** [comment]
@cgivre
Ok~ I will add a test case later

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/primitive/HiveTimestampWriter.java [hunk 1]
```diff
@@ -33,10 +34,14 @@ public HiveTimestampWriter(PrimitiveObjectInspector inspector, TimeStampWriter w
 
   @Override
   public void write(Object value) {
-    String timestampString = PrimitiveObjectInspectorUtils.getString(value, inspector);
-    long timestampMillis = new DateTime(Timestamp.valueOf(timestampString).getTime())
-        .withZoneRetainFields(DateTimeZone.UTC).getMillis();
-    writer.writeTimeStamp(timestampMillis);
+    if (value instanceof LongWritable) {
+      writer.writeTimeStamp(((LongWritable) value).get() / 1000);
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/primitive/HiveTimestampWriter.java:38]:
```suggestion
      writer.writeTimeStamp(((LongWritable) value).get() / DateTimeConstants.MILLIS_PER_SECOND);
```

Comment (cdmikechen) [reply contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/writers/primitive/HiveTimestampWriter.java:38]: @vdiravka 
Thank you for your suggestion ~

**Comment by cdmikechen** [comment]
@cgivre
Hi~ I have added a test case and update codes to newest. You can see whether the current test case is ok.

**Review by cgivre — APPROVED** [review]
LGTM +1

**Pull request merged**